### PR TITLE
Closes #4470 Wire up UI for page action support.

### LINKED ISSUE #4470: Add engine API for pageAction support
This is similar to `browserAction` (https://github.com/mozilla-mobile/android-components/issues/4469) support, and also covered by https://github.com/mozilla-mobile/android-components/issues/4469.

Also involves hooking up a delegate to the web extension object and integrating with the toolbar.

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/concept/toolbar/src/main/java/mozilla/components/concept/toolbar/Toolbar.kt [hunk 1]
```diff
@@ -101,13 +101,21 @@ interface Toolbar {
     fun addBrowserAction(action: Action)
 
     /**
-     * Removes a previously added action (see [addBrowserAction]). If the the provided
+     * Removes a previously added page action (see [addBrowserAction]). If the the provided
```
Comment (csadilek) [inline components/concept/toolbar/src/main/java/mozilla/components/concept/toolbar/Toolbar.kt]: nit: browser

File: components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt [hunk 2]
```diff
@@ -52,41 +54,71 @@ class WebExtensionBrowserMenu internal constructor(
...
         }
+
+        private fun addOrUpdateAction(
+            extension: WebExtensionState,
+            globalAction: Action,
+            tabAction: Action?,
+            menuItems: ArrayList<BrowserMenuItem>,
+            isPageAction: Boolean = false
+        ): Boolean {
+            // Add the global browser/page action if it doesn't exist
```
Comment (csadilek) [inline components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt]: nit: This comment should go on line 107 instead.

**Comment by Amejia481** [comment]
Updated!

**Comment by Amejia481** [comment]
bors r=csadilek

**Comment by Amejia481** [comment]
bors retry

**Comment by Amejia481** [comment]
bors retry

**Comment by Amejia481** [comment]
bors retry

**Comment by Amejia481** [comment]
bors retry

**Comment by Amejia481** [comment]
bors retry

**Pull request merged**